src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Methodologies/MethodologyService.cs

```diff
@@ -79,6 +85,19 @@ public class MethodologyService : IMethodologyService
                         Owner = false
                     });
 
+                    // If the adopted methodology is unpublished, it may require publishing,
+                    // if the adopting publication is live
+                    var versionToBePublishedId = await _methodologyVersionRepository.IsToBePublished(methodology);
```

Comment:
I suppose this method should be called `GetMethodologyVersionToBePublished()`?

Reply:
I wonder if this whole block of code (which sets the published date and publishes the files) should be moved into MethodologyApprovalService and called from here.  See the already- similar logic from MethodologyApprovalService.UpdateStatus():

```
                        // We cannot rely on Methodology.LatestPublishedVersionId, as it may now be incorrect,
                        // if we are approving and publishing this methodology version.
                        var isToBePublished = await _methodologyVersionRepository.IsToBePublished(methodologyVersion);

                        if (isToBePublished)
                        {
                            methodologyVersion.Published = DateTime.UtcNow;
                            methodologyVersion.Methodology.LatestPublishedVersionId = methodologyVersion.Id;

                            await _publishingService.PublishMethodologyFiles(methodologyVersion.Id);
                        }
```